### PR TITLE
refactor: transactions view components

### PR DESCRIPTION
**Pull request by dohaki**
## Needs QA

**Overview**

While working on the “All Transactions” view, I had some problems understanding and reusing the existing code. This PR, therefore, tries to improve that by implementing some best practices
- replace `createTransactionTableJSX` and `createMobileTransactionTableJSX` with proper JSX components
  - it was very hard to understand how the `TransactionsTable` should be used or how the rows look like
  - we now use rows and cells components that in a more declarative way describe the ui
- localize ui state where possible
  - we had a lot of state/handlers in the top-level component/hook, even though they could be moved down the hierarchy
- extract hooks and components into separate functions
- improve separation of concerns and abstractions
- improve folder/file structure

FYI, draft PR for all transactions view that is based on this branch https://github.com/across-protocol/frontend-v2/pull/221.

**Acceptance Criteria**

Use the Vercel preview link and visit the `/transactions` page.

1. See and verify that everything works and looks as before on desktop.
2. See and verify that everything works and looks as before on mobile.

### How To Report QA
Testers, use the following templates to pass the pull request or to report issues that require a resolution.
Following these guidelines will help the developer respond more quickly to issues, but you are free to comment
in any way as needed to help resolve issues.

**Reporting QA Issues**

Testers should use Github comments to report issues with the PR. Use screenshots if relevant, you can also
check the [developer console](https://support.airtable.com/hc/en-us/articles/232313848-How-to-open-the-developer-console)
to see if there are any logs which might be related. Use the following template when reporting a problem or issue with the PR.
You submit multiple comments using this template if there are multiple issues.

```
**QA Issue**

<!--
Describe the problem with you ran into, if it has to do with one of the acceptance criteria, then add it here.
Add any screenshots of the app or console logs if you think it would be useful.
-->

**Repro Steps**

<!--
If this issue was seen after taking particular steps, then note how to do it here.

Example:
1. Go to bridge tab and send a tx between any chains.
2. go to pool tab before going to transaction tab.
3. see that my pending transaction is missing.
-->

**Testing Environment**

<!--
Optionally include anything about your testing environment, ie phone, tablet, chrome or safari, if you used
metamask or a different wallet, etc. This will let other testers know if there are configurations they could
test differently from yours.

Example:
Used Metamask and the latest version of Safari on a Macbook. 
-->


```

**Reporting QA Regression**

A regression is something that was working, but now is no longer working due to the new code changes. The breakage
may seem completely unrelated to the pull request, which is why they can go unnoticed during testing.
If you find a regression, or arent sure, report it anyway as an issue and the engineer can decide how to resolve it. 

**Reporting QA Passed**

If you tested and did not find any issues, you will need to signal to the developer that this passed your QA.
Do this by adding comments in the pr. You should communicate to both the developers and other testers what you did during testing. Use the following
template to signal that QA passed your testing.

```
**QA Pass**

<!--
Describe what you tested, if you tested beyond what was originally specified, or any other parts of the app.
This will help others decide how to do their testing, and let the engineer know the testing coverage of the app.

Example:
Tested sending eth and weth between optimism and arbitrum on kovan and back again. Both cases I was able to see my 
transaction and speed it up. No issues with the UI but I wasnt sure if the transaction actually was faster. Also
tested adding and removing liquidity from the pool tab which worked as expected. 

-->

**Testing Environment**

<!--
Optionally include anything unique about your testing environment, ie phone, tablet, chrome or safari, if you used
Metamask or a different wallet, etc. This will let other testers know if there are configurations they could
test differently from yours.

Example:
Used Metamask and the latest version of Safari on a Macbook. 
-->

```

### CODE REVIEW

File: src/views/Transactions/hooks/useTxClient.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+          );
+          setFilledTransferTuples((prevFilledTransfers) =>
+            getUpdatedTransferTuples(prevFilledTransfers, nextFilledTransfers)
+          );
+          setPendingTransferTuples((prevPendingTransfers) =>
+            getUpdatedTransferTuples(prevPendingTransfers, nextPendingTransfers)
+          );
+        }
+      );
+    }
```
Comment (daywiss) [inline src/views/Transactions/hooks/useTxClient.ts:47]: this need a unsubscribe cleanup?

File: src/views/Transactions/hooks/useTxClient.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      if (account && txClient) {
+        txClient.stopFetchingTransfers(account);
+        setFilledTransferTuples([]);
+        setPendingTransferTuples([]);
+      }
+    };
+  }, [account, txClient]);
+
+  useEffect(() => {
+    return () => {
```
Comment (daywiss) [inline src/views/Transactions/hooks/useTxClient.ts:77]: not sure why this isnt just included in the previous use effect where the timer is set

**Review by daywiss — COMMENTED** [review]
i think this is an improvement over prev implementation as code is smaller and more composable, but we just need to be careful of sweeping refactors and make sure we have a good justification.  this is also fairly high risk as it could break a major feature on the app, so we should be doing rigorous QA.  in my testing i didnt see any issues though, so looks good to me, just a couple nits

File: src/views/Transactions/hooks/useTxClient.ts (hunk 1 above)
Comment (dohaki) [reply src/views/Transactions/hooks/useTxClient.ts:47]: The `TransfersHistoryClient` of the SDK doesn't seem to provide methods to remove event listeners / unsubscribe AFAICT 🤔 There is only the `stopFetchingTransfers` method which we call in the clean-up effect below.

File: src/views/Transactions/hooks/useTxClient.ts (hunk 2 above)
Comment (dohaki) [reply src/views/Transactions/hooks/useTxClient.ts:77]: Most of the code was copy&pasted so not 100% sure. But yea I could move this clean-up to the previous hook

**Review by Tulun — APPROVED** [review]
Seems good to me.

Echoing David's comment this is a bit of a departure from our previous structure but I'm more inclined for this kind of folder structure.

We might want to be very cautious doing this in other parts of the app though. TX page is a read only thing, if this had been done on Pool or Bridge might be a lot of pushback. A refactor may not be crazy but I'd check in first, and we'd need to do some major QA.

**Review by daywiss — APPROVED** [review]

**Review by amateima — APPROVED** [review]

**Pull request merged**